gw_eccentricity/eccDefinition.py

```diff
@@ -654,11 +697,97 @@ def process_data_dict(self,
                         :index_num_orbits_earlier_than_merger]
-                    newDataDict["t"] = newDataDict["t"][
-                        :index_num_orbits_earlier_than_merger]
+            newDataDict["t"] = newDataDict["t"][
+                :index_num_orbits_earlier_than_merger]
         return newDataDict, t_merger, amp_gw_merger, min_width_for_extrema
 
-    def get_amp_phase_omega_gw(self, data_dict):
+    def rotate_modes(self, data_dict):
+        """"Rotate intertial frame modes to obtain coprecessing frame modes.
+        
+        Parameters
+        ----------
+        data_dict: dict
+            Dictionary of modes in the inertial frame. The modes are rotated
+            using `get_coprecessing_data_dict` to obtain the corresponding
+            modes in the coprecessing frame.
+
+            To obtain coprecessing frame modes from an inertial frame dictionary,
+            this dictionary should include all (ell, m) modes for a specified
+            ell value. For instance, the inertial modes dictionary should
+            contain at least all the modes for `ell=2`, i.e., (2, -2), (2, -1),
+            (2, 0), (2, 1), and (2, 2), to achieve accurate coprecessing frame
+            modes.
+
+        Returns
+        -------
+        data_dict with the inertial modes replaced by the corresponding
+        modes in the coprecessing frame.
+        """
+        if "hlm" in data_dict:
+            data_dict = get_coprecessing_data_dict(data_dict)
+        if "hlm_zeroecc" in data_dict:
+            # `get_coprecessing_data_dict` looks for modes using the key "hlm"
+            # and the associated time using "t".
+            # Make a dictionary with a key "hlm" and pass the zeroecc modes 
+            # dict via it. Similary for the times using "t".
+            intertial_zeroecc_data_dict = {"t": data_dict["t_zeroecc"],
+                                           "hlm": data_dict["hlm_zeroecc"]}
+            coprecessing_zeroecc_data_dict = get_coprecessing_data_dict(
+                intertial_zeroecc_data_dict)
+            # update the data_dict by replacing the inertial frame zeroecc
+            # modes with coprecessing frame zeroecc modes.
+            data_dict.update(
+                {"hlm_zeroecc": coprecessing_zeroecc_data_dict["hlm"]})
+
+        # if hlm is not in data_dict, get it from amplm and phaselm.
+        # We need to provide hlm to get the rotated modes.
+        if "hlm" not in data_dict:
+            amplm_dict = self.get_amplm_from_dataDict(data_dict)
+            phaselm_dict = self.get_phaselm_from_dataDict(data_dict)
+            hlm_dict = {}
+            # check if "hlm_zeroecc" is not in data_dict but amplm_zeroecc is
+            # in data_dict
+            if "amplm_zeroecc" in data_dict and "hlm_zeroecc" not in data_dict:
+                add_hlm_zeroecc = True
+                hlm_zeroecc_dict = {}
+            else:
+                add_hlm_zeroecc = False
```

Comment:
Again, seems very convoluted to do handle zeroecc this way. Can it be done more simply using tags like above?